### PR TITLE
byCrookie.TypeCode version 1.1.11

### PR DESCRIPTION
**Pull request by byCrookie**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/96967)

### CODE REVIEW

**Comment by byCrookie** [comment]
@microsoft-github-policy-service agree company="byCrookie"

**Comment by byCrookie** [comment]
The installer validation probably fails because it has no certificate. Is there a way to get more details for the error?

File: manifests/b/byCrookie/TypeCode/1.1.9/byCrookie.TypeCode.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.4.0.schema.json
+
+PackageIdentifier: byCrookie.TypeCode
+PackageVersion: 1.1.9
+InstallerType: msi
+InstallModes:
+  - silent
```
Comment (mdanish-kh) [inline manifests/b/byCrookie/TypeCode/1.1.9/byCrookie.TypeCode.installer.yaml:8]:
```suggestion
  - silent
AppsAndFeaturesEntries:
  - DisplayVersion: 1.0.0.0
```

**Review by mdanish-kh — CHANGES_REQUESTED** [review]
`DisplayVersion` written to registry (that you can see in Control Panel) is `1.0.0.0`. I'm assuming this is a mistake in the installer.

WinGet relies on metadata from Control Panel / Apps&Features to match with the manifest for upgrade scenario. Either the installer needs to be fixed to write the correct version to registry, or if that's not possible a temporary workaround would be to use `AppsAndFeaturesEntries` field in the manifest and explicitly provide the DisplayVersion key.

I would highly recommend the former approach as adding AppsAndFeaturesEntries like this also has a potential to cause infinite upgrade loop scenarios for the user when a new version for the package would be added to the repo

**Comment by mdanish-kh** [comment]
[Policy] Installer-Issue

File: manifests/b/byCrookie/TypeCode/1.1.9/byCrookie.TypeCode.installer.yaml (hunk 1 above)
Comment (byCrookie) [reply manifests/b/byCrookie/TypeCode/1.1.9/byCrookie.TypeCode.installer.yaml]: Thank you for the display version hint. I have updated the msi installer to include the display version.

File: manifests/b/byCrookie/TypeCode/1.1.11/byCrookie.TypeCode.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.4.0.schema.json
+
+PackageIdentifier: byCrookie.TypeCode
+PackageVersion: 1.1.11
+InstallerType: msi
+InstallModes:
+  - silent
```
Comment (mdanish-kh) [inline manifests/b/byCrookie/TypeCode/1.1.11/byCrookie.TypeCode.installer.yaml:8]:
```suggestion
  - silent
AppsAndFeaturesEntries:
  - DisplayVersion: 1.1.11.0
```

I see that you market the version as `1.1.11` in your GitHub Releases, but WinGet requires the version in Registry / Control Panel / Apps&Features for matching upgrades. One resolution is to use the "actual" DisplayVersion field of Registry in place of the PackageVersion field in the manifest.
Another resolution is to use the "marketing" version in the PackageVersion field (this version will be shown to the user for winget install), but use `AppsAndFeaturesEntries` and explicitly provide the "actual" DisplayVersion for WinGet to match upgrades. I've suggested the latter here.

File: manifests/b/byCrookie/TypeCode/1.1.11/byCrookie.TypeCode.installer.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+Installers:
+  - InstallerUrl: https://github.com/byCrookie/TypeCode/releases/download/v1.1.11/TypeCode.Wpf.Setup.msi
+    Architecture: x64
+    InstallerSha256: b413e5f0f4460a406225f1ee07d43aa71ac6148aa79f559f134960436c88f12f
```
Comment (mdanish-kh) [inline manifests/b/byCrookie/TypeCode/1.1.11/byCrookie.TypeCode.installer.yaml:12]:
```suggestion
    InstallerSha256: b413e5f0f4460a406225f1ee07d43aa71ac6148aa79f559f134960436c88f12f
    ProductCode: '{C7465337-21EF-45C1-A6CC-2101884FF304}'
```
ProductCode field of MSI Installers for better upgrade matching.

File: manifests/b/byCrookie/TypeCode/1.1.11/byCrookie.TypeCode.installer.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    InstallerSwitches:
+      Silent: /quiet
+      SilentWithProgress: /passive
```
Comment (mdanish-kh) [inline manifests/b/byCrookie/TypeCode/1.1.11/byCrookie.TypeCode.installer.yaml:15]:
```suggestion
```
These switches are not needed as they are automatically passed to the installer with some additional switches if the InstallerType is set to msi.

File: manifests/b/byCrookie/TypeCode/1.1.11/byCrookie.TypeCode.locale.en-US.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.4.0.schema.json
+
+PackageIdentifier: byCrookie.TypeCode
+PackageVersion: 1.1.11
+PackageLocale: en-US
+Publisher: byCrookie
```
Comment (mdanish-kh) [inline manifests/b/byCrookie/TypeCode/1.1.11/byCrookie.TypeCode.locale.en-US.yaml:7]:
```suggestion
Publisher: byCrookie
PublisherUrl: https://github.com/byCrookie/
PublisherSupportUrl: https://github.com/byCrookie/TypeCode/issues
Author: byCrookie
```
Additional metadata

File: manifests/b/byCrookie/TypeCode/1.1.11/byCrookie.TypeCode.locale.en-US.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.4.0.schema.json
+
+PackageIdentifier: byCrookie.TypeCode
+PackageVersion: 1.1.11
+PackageLocale: en-US
+Publisher: byCrookie
+PackageName: TypeCode
```
Comment (mdanish-kh) [inline manifests/b/byCrookie/TypeCode/1.1.11/byCrookie.TypeCode.locale.en-US.yaml:8]:
```suggestion
PackageName: TypeCode
PackageUrl: https://github.com/byCrookie/TypeCode
```
Additional metadata

File: manifests/b/byCrookie/TypeCode/1.1.11/byCrookie.TypeCode.locale.en-US.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.4.0.schema.json
+
+PackageIdentifier: byCrookie.TypeCode
+PackageVersion: 1.1.11
+PackageLocale: en-US
+Publisher: byCrookie
+PackageName: TypeCode
+License: GPL-3.0
```
Comment (mdanish-kh) [inline manifests/b/byCrookie/TypeCode/1.1.11/byCrookie.TypeCode.locale.en-US.yaml:9]:
```suggestion
License: GPL-3.0
LicenseUrl: https://github.com/byCrookie/TypeCode/blob/main/LICENSE
```
Additional metadata

File: manifests/b/byCrookie/TypeCode/1.1.11/byCrookie.TypeCode.locale.en-US.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.4.0.schema.json
+
+PackageIdentifier: byCrookie.TypeCode
+PackageVersion: 1.1.11
+PackageLocale: en-US
+Publisher: byCrookie
+PackageName: TypeCode
+License: GPL-3.0
+ShortDescription: Develop c# code faster by generating .NET specific boilerplate code using reflection on assemblies.
```
Comment (mdanish-kh) [inline manifests/b/byCrookie/TypeCode/1.1.11/byCrookie.TypeCode.locale.en-US.yaml:10]:
```suggestion
ShortDescription: Develop c# code faster by generating .NET specific boilerplate code using reflection on assemblies.
Moniker: typecode
Tags:
- boilerplate-generation
- code-generation
- console-application
- csharp
- netcore
- wpf-application
ReleaseNotesUrl: https://github.com/byCrookie/TypeCode/releases/tag/v1.1.11
Documentations:
- DocumentLabel: Wiki
  DocumentUrl: https://github.com/byCrookie/TypeCode/wiki
```
Additional metadata

**Review by mdanish-kh — CHANGES_REQUESTED** [review]

**Review by mdanish-kh — APPROVED** [review]

**Comment by mdanish-kh** [comment]
[Policy] reset feedback

**Pull request merged**